src/api-service/__app__/onefuzzlib/azure/vm.py

```diff
@@ -248,10 +259,11 @@ def create(self) -> Union[None, Error]:
             self.image,
             self.auth.password,
             self.auth.public_key,
+            self.nsg,
         )
 
     def delete(self) -> bool:
-        return delete_vm_components(str(self.name))
+        return delete_vm_components(str(self.name), self.nsg)
```

Comment:
If there is one NSG per region, and now multiple debug VMs get spun up, then 1 gets deleted, won't cause issues?